docs/api.rst

```diff
@@ -449,8 +454,182 @@ with instantaneous transition rate from ``i`` to ``j`` that is equal to
 ``rate`` multiplied by ``transition_matrix[i,j]``.
 The ``root distribution`` and every row in the ``transition_matrix``
 must give *probabilities*, i.e., they must be nonnegative numbers summing to 1.
+For the precise interpretation of these parameters
+(especially when the transition matrix has nonzero entries on the diagonal)
+see :ref:`sec_api_mutation_matrix_theory`.
+
+.. autoclass:: msprime.MatrixMutationModel
+
+You can define your own, but you probably don't need to:
+there are several mutation matrix models already implemented in msprime,
+using binary (0/1), nucleotide, or amino acid alphabets:
+
+
+.. autoclass:: msprime.BinaryMutations
+
+.. autoclass:: msprime.JukesCantor
+
+TODO: move to msprime/mutations.py
+
+The Jukes-Cantor 1969 mutation model, where each nucleotide mutates to each
+other nucleotide at the same rate.
+
+This is a :class:`.MatrixMutationModel` with alleles ``["A", "C", "G", "T"]``,
+root distribution ``[1/4, 1/4, 1/4, 1/4]``, and transition matrix
+``[[0, 1/3, 1/3, 1/3], [1/3, 0, 1/3, 1/3], [1/3, 1/3, 0, 1/3], [1/3, 1/3, 1/3, 0]]``.
+It has no free parameters.
+
+Citation: *Jukes TH, Cantor CR (1969). Evolution of Protein Molecules. pp. 21-132.*
+
+.. autoclass:: msprime.HKY
+
+TODO: move to msprime/mutations.py
+
```

Comment:
Ok, the following is correct now, I believe. Note that the discussion below is somewhat complicated by the scaling factor `M`, required to make `P` a valid transition matrix. It looks like seq-gen avoids this issue by not documenting what the overall mutation rate actually is. Do you know, @GertjanBisschop - what's the overall mutation rate for seq-gen under HKY?  Are we testing for equality of total number of mutations or just relative frequencies of different transition types?  And, how does this description seem to you - clear? correct?

We *could* make the discussion below so that the mutation rates are just `mu * q[i,j]`, if we added a "rate scaling factor" to the mtuation model object, and put M in there, transparent to the user. But I think this is probably not worth it - what do you think?

Reply:
This is great work. Will review this and answer all your questions tonight. Whichever way we decide on the scaling factor `M`, we will have one up on SeqGen given that this is all now very well documented.

Reply:
> But I would argue against adding the rescaling factor to the mutation model object. I

Oh, good. =)

> If I understand things correctly, you would never allow the user, defining their own MatrixMutationModel to specify any M value, right?

The way I'd imagine it would be that the MatrixMutationModel would have a `rate_scaling` argument ("M"), that gets multiplied by the mutation rate when it is used with mutate( ). So, yes, the user would be able to set this, but only if they are doing their own model. Then, if we defined HKY as a MatrixMutationModel(... rate_scaling=M), then we'd be able to delete all references to M from the documentation here.